### ISSUE TITLE
Deploy-ng distributable build fails, reporting "optimize" being an invalid keyword

### ISSUE THREAD

**Issue by ArsThaumaturgis** [panda3d/panda3d#388]
In attempting to build a distributable via Deploy-ng for #277, I ran into a problem: Running deploy-ng (via "python setup.py bdist_apps") results in the following error:
`Traceback (most recent call last):
  File "setup.py", line 17, in <module>
    "manylinux1_x86_64",
  File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python2.7/dist-packages/direct/showutil/dist.py", line 1131, in run
    self.run_command('build_apps')
  File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/usr/local/lib/python2.7/dist-packages/direct/showutil/dist.py", line 293, in run
    self.build_runtimes(platform, use_wheels)
  File "/usr/local/lib/python2.7/dist-packages/direct/showutil/dist.py", line 567, in build_runtimes
    create_runtime(appname, scriptname, False)
  File "/usr/local/lib/python2.7/dist-packages/direct/showutil/dist.py", line 547, in create_runtime
    }, self.log_append)
  File "/usr/local/lib/python2.7/dist-packages/direct/showutil/FreezeTool.py", line 1766, in generateRuntimeFromStub
    code = compile(code, moduleName, 'exec', optimize=2)
TypeError: 'optimize' is an invalid keyword argument for this function`

Believe that I have a recent build of the "deploy-ng" branch of the engine installed via "pip". Specifically, I believe that I downloaded it yesterday (Friday the 7th of September).

**Comment by rdb** [comment]
I'm assuming you are building with Python 2.  If so, I guess the optimize option that is used is not supported by Python 2.  The easiest way to work around this is simply to remove the `, optimize=2` bit from that line for now.

**Comment by Moguri** [comment]
Looks like optimize was added in 3.2. Assigning this to myself to fix.

**Comment by ArsThaumaturgis** [comment]
Hmm... "pip" doesn't want to install the wheel from the download page. I'm getting the following error:
`panda3d-1.10.0.dev1756+deployng.236-cp35-cp35m-manylinux1_x86_64.whl is not a supported wheel on this platform.
`

I'm downloading from the Ubuntu Xenial page, using [this URL](https://www.panda3d.org/download.php?platform=xenial&version=devel&sdk&branch=deploy-ng).

A few days ago I installed a version downloaded from the same page, I believe, and that installed successfully.

The only apparent difference that I note is that the successful version had the string "cp27-cp27mu", while this one has "cp35-cp35m". At a guess, that's not Python 3, is it? I'm using Python 2, I believe, which might be the reason that it's not installing.

(I also note that one ends in "m", while the other ends in "mu"--not knowing what those indicate, I don't know how relevant that is.)

I'm intend to build the engine from source in order to make a fork and provide a "pull request" for #340, so I should hopefully be able to test this issue from that build.

Either way, I see no "27" builds in the past few entries on that page--will the dev. version of Panda no longer support Python 3, then?

**Comment by rdb** [comment]
Yes, you are attempting to install a Python 3.5 build into a Python 2.7 installation, hence the error.

I have poked the buildbot and it has now updated the page with Python 2.7 builds.

For what it's worth, the easiest way to make sure you have the proper build of the latest version of the deploy-ng branch build is just to run this command:
```
pip install --pre --extra-index-url https://archive.panda3d.org/branches/deploy-ng panda3d
```
Although, in this case, it would have installed an older build since no Python 2.7 build was available of the latest version, before I had new builds generated.

**Comment by ArsThaumaturgis** [comment]
Ah, fair enough, and thank you for poking the buildbot! I've just downloaded the new build, I believe.

I didn't know about that pip command--thank you for mentioning it! ^_^

That said, I think that in cases like this, where a build after a specific commit is called for, it might be safer to stick to explicitly downloading the build. That way, one is more likely to catch issues such as the one that I ran into, and less likely to end up testing with an old version, as you've noted would presumably have been the case had I used that command in this instance.

Since I haven't yet gotten around to building the engine from the fork that I made for #340, I think that I'll leave that, and simply test with the build that I've now downloaded.
